src/node/txreconciliation.cpp

```diff
@@ -197,6 +210,86 @@ class TxReconciliationTracker::Impl
+                best_peers.insert(std::make_pair(hash_key, indexed_state.first));
+            }
+        }
+
+        std::vector<NodeId> result;
+        auto it = best_peers.begin();
+        for (size_t i = 0; i < targets && it != best_peers.end(); ++i, ++it) {
+            result.push_back(it->second);
+        }
+        return result;
+    }
+
+    bool ShouldFanoutTo(const uint256& wtxid, CSipHasher deterministic_randomizer, NodeId peer_id,
+                        size_t inbounds_nonrcncl_tx_relay, size_t outbounds_nonrcncl_tx_relay)
+                        const EXCLUSIVE_LOCKS_REQUIRED(!m_txreconciliation_mutex)
+    {
+        AssertLockNotHeld(m_txreconciliation_mutex);
+        LOCK(m_txreconciliation_mutex);
+        if (!IsPeerRegistered(peer_id)) return true;
+        // We use the pre-determined randomness to give a consistent result per transaction,
+        // thus making sure that no transaction gets "unlucky" if every per-peer roll fails.
+        deterministic_randomizer.Write(wtxid.GetUint64(0));
+        const auto& recon_state = std::get<TxReconciliationState>(m_states.find(peer_id)->second);
+        // We decide whether a particular peer is a low-fanout flood target differently
+        // based on its connection direction:
+        // - for outbounds we have a fixed number of flood destinations;
+        // - for inbounds we use a fraction of all inbound peers supporting tx relay.
+        //
+        // We first decide how many reconciling peers of a given direction we want to flood to,
+        // and then generate a list of peers of that size for a given transaction. We then see
+        // whether the given peer falls into this list.
+        double destinations;
+        if (recon_state.m_we_initiate) {
+            destinations = OUTBOUND_FANOUT_DESTINATIONS - outbounds_nonrcncl_tx_relay;
+        } else {
+            const size_t inbound_rcncl_peers = std::count_if(m_states.begin(), m_states.end(),
+                                                        [](std::pair<NodeId, std::variant<uint64_t, TxReconciliationState>> indexed_state) {
+                                                            const auto* cur_state = std::get_if<TxReconciliationState>(&indexed_state.second);
+                                                            if (cur_state) return !cur_state->m_we_initiate;
+                                                            return false;
+                                                        });
+
+            // Since we use the fraction for inbound peers, we first need to compute the total
+            // number of inbound targets.
+            const double inbound_targets = (inbounds_nonrcncl_tx_relay + inbound_rcncl_peers) * INBOUND_FANOUT_DESTINATIONS_FRACTION;
+            destinations = inbound_targets - inbounds_nonrcncl_tx_relay;
+        }
+
+        if (destinations < 0.01) {
```

Comment:
Right, got you. I guess we can also ask this the other way around to see if there is a cleaner workaround. In what case is `destinations` smaller than 1 but we still want to pass it to `GetFanoutTargets`?

Reply:
Say we have 5 inbound Erlay peers and 0 inbound legacy peers.
`inbound_targets = (5+0) * 0.1 = 0.5`
`destinations = 0.5 - 0 = 0.5`

It just means that we will take 1 inbound peer for fanout with a 50% chance.

I guess it will work just fine if i drop the `0.01` (1% or less chance) check. It's kinda unfortunate we have to do the compute if chance is that small. Fine with me either way, let me know what you think.

Reply:
Just adding a comment with some context, saying that for chances lower than 1% it may not be worth the computation may be fine. It just struck me as weird where that value came from.